.github/workflows/go-test-integration.yaml

```diff
@@ -0,0 +1,46 @@
+name: Go Test Integration
+on:
+  workflow_call:
+    inputs:
+      services:
+        required: false
+        type: string
+        description: The service(s) to watch and abort on for the run of the integration test
+        default: integration-test
+
+jobs:
+  integration-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Go mod cache
+        id: go-cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cache/go-build
+            vendor
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.*') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+      - name: Go mod vendor
+        if: steps.go-cache.outputs.cache-hit != 'true'
+        env:
+          GOPRIVATE: "github.com/encodium"
+        run: |
+          echo "machine github.com login rp-build-user password ${{ secrets.CLONE_PAT }}" > ~/.netrc
```

Comment:
If I remember correctly, secrets like this need to be passed in from the caller as an input.. they can't just be used.

Reply:
whoops, that is correct